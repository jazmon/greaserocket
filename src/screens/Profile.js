// @flow
import React from 'react';
import {
  View,
  Text,
  Platform,
  StyleSheet,
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

type Props = {
  navigation: Object,
};

type State = {
  text: string,
};

class Profile extends React.Component {
  static navigationOptions = {
    tabBar: {
      label: 'Profile',
      icon: ({ tintColor }) => (
        <Icon
          name="ios-person"
          size={30}
          color={tintColor}
        />
     ),
    },
  }

  props: Props;

  constructor(props: Props) {
    super(props);

    this.state = {
      text: 'Profile',
    };
  }

  state: State;

  render() {
    const { text } = this.state;
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{text}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: '#000',
    fontSize: 16,
  },
});

export default Profile;
