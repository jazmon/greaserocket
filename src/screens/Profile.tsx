// @flow
import React from 'react';
import { View, ActivityIndicator } from 'react-native';
import moment from 'moment';
import Icon from 'react-native-vector-icons/Ionicons';
import { connect } from 'react-redux';
import styled from 'styled-components/native';

import { ReduxState } from 'redux/modules';
import { TestComponent } from 'components/TestComponent';
import { Maybe } from 'types';
import { Dispatch } from 'redux';

// import type {
//   NavigationScreenProp,
//   NavigationAction
// } from 'react-navigation/lib/TypeDefinition';

import { login } from 'redux/modules/user';
import { ThemeType } from 'constants/theme';

interface Props {
  // navigation: NavigationScreenProp<*, NavigationAction>,
  login(): void;
  loginDate: Date;
  profile: Maybe<Auth0Profile>;
  loading: boolean;
  token: Maybe<Auth0Token>;
}

const Container = styled.View`
  flex-grow: 1;
  flex-direction: column;
  background-color: ${({ theme }: { theme: ThemeType }) => theme.colors.white};
  align-items: center;
  justify-content: center;
`;

const LoadingContainer = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const BaseText = styled.Text`
  color: ${({ theme }: { theme: ThemeType }) => theme.text.colors.primary};
  font-size: 16;
`;

class Profile extends React.Component<Props, void> {
  static navigationOptions = {
    tabBarLabel: 'Profile',
    tabBarIcon: ({ tintColor }: { tintColor: string }) =>
      <Icon name="ios-person" size={30} color={tintColor} />,
  };

  componentDidMount() {
    setTimeout(() => {
      // FIXME
      if (
        !this.props.profile ||
        moment(this.props.loginDate).isBefore(moment().subtract(20, 'seconds'))
      ) {
        this.props.login();
      }
    }, 400);
  }

  render() {
    console.log('token', this.props.token);
    const { profile, loading } = this.props;
    return (
      <Container>
        {loading &&
          <LoadingContainer>
            <ActivityIndicator size="large" />
          </LoadingContainer>}
        {profile &&
          <View>
            <TestComponent />
            <BaseText>
              {profile.name}
            </BaseText>
            <BaseText>
              {profile.email}
            </BaseText>
          </View>}
      </Container>
    );
  }
}

const mapState = ({ user }: ReduxState) => ({ ...user });

const mapActions = (dispatch: Dispatch<ReduxState>) => ({
  login: () => dispatch(login()),
});

export const ProfileComponent = Profile;
export default connect(mapState, mapActions)(Profile);
