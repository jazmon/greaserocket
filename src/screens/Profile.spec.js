import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { ProfileComponent as Profile } from './Profile';

it('renders correctly', () => {
  const tree = renderer.create(<Profile login={jest.fn} />);
  expect(tree).toMatchSnapshot();
});
