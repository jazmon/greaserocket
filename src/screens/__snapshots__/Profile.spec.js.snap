exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontSize": 16,
      }
    }>
    Profile
  </Text>
</View>
`;
